reframe/core/runtime.py

```diff
@@ -176,14 +176,17 @@ class RuntimeContext:
 
     This class essentially groups the current host system and the associated
     resources of the framework on the current system.
+    It also encapsulates other runtime parameters that are relevant to the
+    framework's execution.
 
     There is a single instance of this class globally in the framework.
 
     .. note::
        .. versionadded:: 2.13
+
     """
 
-    def __init__(self, dict_config, sysdescr=None):
+    def __init__(self, dict_config, sysdescr=None, **kwargs):
```

Comment:
I just wonder if this is the way we should grow this interface. 
Does it make sense to have a parameter for this option? Because `**kwargs` are a bit demanding to reason about, especially in the future when we will forget about this code.

Reply:
`kwargs` is a bad name, I should change it, but I want to keep this for the following reasons:

1. If I just had arguments, I wouldn't force the callers to pass them as key/value pairs.
2. It eliminates duplication also at the interface level.
3. Since these are general options, I wouldn't bother on how I pass them here. There is a clean API (through the `RuntimeContext`) that defines how to access them and this is the only way for the rest of the code to use them. So I don't think it creates confusion.